src/components/WalletModal/index.tsx

```diff
@@ -279,18 +279,28 @@ export default function WalletModal({
     })
   }
 
-  function addNetwork() {
+  async function addNetwork() {
     const chainID: ChainId = NETWORK_CHAIN_ID
-    injected.getProvider().then(provider => {
-      provider
-        .request({
-          method: 'wallet_addEthereumChain',
-          params: [CHAIN_PARAMS[chainID]]
-        })
-        .catch((error: any) => {
-          console.log(error)
-        })
-    })
+    const provider = await injected.getProvider()
+    try {
+      await provider.request({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId: CHAIN_PARAMS[chainID].chainId }]
+      })
+    } catch (switchError) {
+      // This error code indicates that the chain has not been added to MetaMask.
+      if ((switchError as any)?.code === 4902) {
+        try {
+          await provider.request({
+            method: 'wallet_addEthereumChain',
+            params: [CHAIN_PARAMS[chainID]]
+          })
+        } catch (addError) {
+          console.log(addError)
+        }
+      }
+      console.log(switchError)
+    }
```

Comment:
I feel like MM docs nest the code in a weird way, do you think this is cleaner?

```suggestion
    try {
      try {
        await provider.request({
          method: 'wallet_switchEthereumChain',
          params: [{ chainId: CHAIN_PARAMS[chainID].chainId }]
        })
      } catch (e) {
        // This error code indicates that the chain has not been added to MetaMask.
        if ((e as any)?.code !== 4902) {
          throw e;
        }
        
        await provider.request({
          method: 'wallet_addEthereumChain',
          params: [CHAIN_PARAMS[chainID]]
        })
      }
    } catch (e) {
      console.error(e)
    }
```

Reply:
This makes it so there's a top-level catch, instead of nesting inside catches.

Basically this:

```
something().then(somethingElse).catch(topLevelCatch)
```

Instead of this:
```
something()
    .catch(
        somethingElse().catch(innerCatch))
     )
```

Reply:
ty!